From 9fec7020491cc5fd3ef8087324e23357f91decad Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@freescale.com>
Date: Fri, 1 Sep 2017 15:36:58 +0800
Subject: [PATCH 2486/5242] MLK-16350-6: ASoC: fsl_esai: fix the slow bit
 clock rate

commit  0c114e50eae32efa10f949e289ca7ca57341fe37 from
https://source.codeaurora.org/external/imx/linux-imx.git

When the mclk is 49M, but hk_rate is 24M, condition for search
the corret pm parameter is wrong. which cause the pm=999, then
the output bit clock rate is very small.

Signed-off-by: Shengjiu Wang <shengjiu.wang@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/fsl_esai.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/fsl/fsl_esai.c b/sound/soc/fsl/fsl_esai.c
index 0a39fa2..6370d59 100644
--- a/sound/soc/fsl/fsl_esai.c
+++ b/sound/soc/fsl/fsl_esai.c
@@ -226,7 +226,7 @@ static int fsl_esai_divisor_cal(struct snd_soc_dai *dai, bool tx, u32 ratio,
 
 			/* Calculate the fraction */
 			sub = sub * 1000 / ratio;
-			if (sub < savesub) {
+			if (sub <= savesub) {
 				savesub = sub;
 				pm = i;
 				fp = j;
-- 
1.7.9.5

